smarts/core/agent_manager.py

```diff
@@ -223,25 +223,27 @@ def observe(
                 else:
                     logging.log(
                         logging.DEBUG,
-                        f"Agent `{agent_id}` has raised done with {observations[agent_id].events}",
+                        # pytype: disable=attribute-error
+                        f"Agent `{agent_id}` has raised done with {observations[agent_id].events}", 
+                        # pytype: enable=attribute-error
                     )
 
-                rewards[agent_id] = vehicle.trip_meter_sensor(increment=True)
-                scores[agent_id] = vehicle.trip_meter_sensor()
+                rewards[agent_id] = float(vehicle.trip_meter_sensor(increment=True))
+                scores[agent_id] = float(vehicle.trip_meter_sensor())
 
         if sim.should_reset:
             dones = {agent_id: True for agent_id in self.agent_ids}
             dones["__sim__"] = True
 
-        return observations, rewards, scores, dones
+        return observations, rewards, scores, dones 
 
-    def _vehicle_reward(self, vehicle_id, sim):
-        return sim.vehicle_index.vehicle_by_id(vehicle_id).trip_meter_sensor(
+    def _vehicle_reward(self, vehicle_id, sim) -> float:
+        return float(sim.vehicle_index.vehicle_by_id(vehicle_id).trip_meter_sensor(
             increment=True
-        )
+        ))
 
-    def _vehicle_score(self, vehicle_id, sim):
-        return sim.vehicle_index.vehicle_by_id(vehicle_id).trip_meter_sensor()
+    def _vehicle_score(self, vehicle_id, sim) -> float:
+        return float(sim.vehicle_index.vehicle_by_id(vehicle_id).trip_meter_sensor())
```

Comment:
Avoid ad-hoc type casting, i.e., `float( ... )` . If there is any issue with the vehicle.trip_meter_sensor() function, then we should correct it at the source.